WordPress/gutenberg
# Make usage of the settings icon more consistent

Fixes https://github.com/WordPress/gutenberg/issues/62873

## What?

This PR only aims to make the usage of the `settings` icon more consistent in the editor UI. This icon should only be used to switch a control to an alternative version, see for example the various margin, padding, spacing controls.

Instead, it was used inconsistently in at least 3 other cases:
- The icon button to open the Fonts Library, now replaced with a button with visible text after https://github.com/WordPress/gutenberg/pull/62129
- The 'View Options' icon button in the data views.
- The 'Display settings' icon button in the Query Loop block toolbar.

Screenshot of the icon, for reference:

![Screenshot 2024-07-01 at 13 46 12](https://github.com/WordPress/gutenberg/assets/1682452/8d081a28-4301-4980-b3ba-edcf27664dd8)

Work on this PR surfaced other issues. Most notably, the Query loop block had the 'Display settings' placed in the toolbar, while it makes more sense to have them in the Inspector together with other similar settings.
Other findings will be tracked in new issues.

## Why?

Icons should uniquely identify specific features or, at least, _similar_ features.
There is already an icon for the concept of 'settings': the `cog` icon is commonly used for this purpose. It makes sense to reuse popular conventions to make the editor UI more intuitive and less disorienting.


## How?

- Moves the Query Loop block 'Display settings' to the inspector.
- Simplifies the  'Items per page', 'Offset', and 'Max pages to show' settings by removing the `BaseControl`s that were wrapping the `NumberControl`s. Not sure why these wrapping `BaseControl`s were used in teh first place but they seem unnecessary. Also, the description of the 'Max pages to show' setting was incorrectly set to the wrapping `BaseControl` instead of the actual `NumberControl`.
- Renames 'Max page to show' (singular 'page') to 'Max pages to show' (plural 'pages')
- Changes the dataviews 'View Options` icon to use the `cog` icon.

## Testing Instructions





### Old testing instructions

- Select a Query Loop block.
- Make sure 'Inherit query from template' in the Inspector is disabled.
- Observe they Query Loop block toolbar doesn't show the 'Display settings' button any longer.
- Observe the 'Items per page', 'Offset', and 'Max pages to show' settings are now in the Inspector.
- Test all the three settings and observe they work as expected.
- Note: the 'Max pages to show' doesn't update the preview in the editor, not even on trunk. I'll create a separate issue for this.
- Observe the 'Max pages to show' description is correctly associated with the number field via `aria-describedby` and ID attributes.
- Go to the Site Editor > Pages
- Observe the 'View Options' icon button now uses the `cog` icon.

### New testing instructions
- Go to the Site Editor > Pages
- Observe the 'View Options' icon button now uses the `cog` icon.
- Select a Query Loop block.
- Make sure 'Inherit query from template' in the Inspector is disabled.
- Click  'Display settings' in the Query Loop block toolbar.
- Observe the UI within the popover doesn't have relevant visual changes.
- Observe the description of the 'Max pages to show' number field is correctly associated to the input with aria-describedby.


### Testing Instructions for Keyboard


## Screenshots or screencast 

The Query loop block settings moved to the Inspector:

![Screenshot 2024-07-01 at 13 50 21](https://github.com/WordPress/gutenberg/assets/1682452/80086093-3167-4773-ae31-43af15a5be27)

The new cog icon in the data views:

![Screenshot 2024-07-01 at 12 43 25](https://github.com/WordPress/gutenberg/assets/1682452/c04aef48-1461-4db3-90e8-52594e8bdc93)
